pkg/configurations/configurations.go

```diff
@@ -0,0 +1,159 @@
+package configurations
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/gobwas/glob"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/thoas/go-funk"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/spf13/viper"
+)
+
+const defaultInterfacesFilter = "veth*"
+const defaultNodeNetworkStateRefresh = 5
+
+var (
+	log   = logf.Log.WithName("configurations")
+	mutex = sync.Mutex{}
+)
+
+// Struct to match config file
+type config struct {
+	NodeNetworkRefreshInterval int    `mapstructure:"node_network_state_refresh_interval"`
+	InterfaceFilter            string `mapstructure:"interfaces_filter"`
+	InterfacesFilterGlob       glob.Glob
+}
+
+var c = config{
```

Comment:
I'm not a big fan of coupling watcher and config. Responsibilities are not so clear here. What about following:

* Split watcher and configurations to two separate files in this package
* Extend config with getters and setters. Maybe one setter to set the whole thing and one getter per each value. All these will employ a mutex.
* Both watcher and user (readers of loaded values) will use these synchronous/safe getters and setters
* With this, the user facing logic (config alone) will be split from the logic of updating it.

Reply:
But config setter is loading yaml to struct. How do u see and I need viper for it. It will look really strange but I'll try.

Reply:
Viper has nothing to do with the fact that reload is dynamic, so it can be part of the config struct. Config can have loadFromYaml/File method

Reply:
hmm need to think about it. 
now i am doing it using viper "err := v.Unmarshal(&c)". It is easy and very easily extendable and u ask me to read yaml.
I really don't think there is any need to be so generic and decoupled.

Reply:
I care about the interface for the rest of the application and decoupling of the daemon which reads the current implementation of the config (ConfigMap as a file) from the config object.

Reply:
got it